lib/elixir/lib/protocol.ex

```diff
@@ -716,6 +718,66 @@ defmodule Protocol do
     end
   end
 
+  def __callback_ast_to_fa__({kind, {:"::", _, [{name, _, args}, _return]}, _pos})
+      when kind in [:callback, :macrocallback] do
+    {name, length(args)}
+  end
+
+  def __warn__(message, env) do
+    IO.warn(message, Macro.Env.stacktrace(env))
+  end
+
+  # TODO: Convert the following warnings into errors future Elixir versions
+  defmacro __before_compile__(env) do
```

Comment:
This should be a function. I believe there is no reason to inject this code into the module.

Reply:
Done.